gcpkms/keysource.go

```diff
@@ -233,14 +234,32 @@ func (key *MasterKey) newKMSClient() (*kms.KeyManagementClient, error) {
 	return client, nil
 }
 
-// getGoogleCredentials returns the SopsGoogleCredentialsEnv variable, as
+// getGoogleCredentials returns the connection options taken from the
+// SopsGoogleCredentialsEnv or SopsGoogleCredentialsOAuthEnv variables, as
 // either the file contents of the path of a credentials file, or as value in
-// JSON format. It returns an error if the file cannot be read, and may return
-// a nil byte slice if no value is set.
-func getGoogleCredentials() ([]byte, error) {
-	defaultCredentials := os.Getenv(SopsGoogleCredentialsEnv)
-	if _, err := os.Stat(defaultCredentials); err == nil {
-		return os.ReadFile(defaultCredentials)
-	}
-	return []byte(defaultCredentials), nil
+// JSON format. It returns an error if the file cannot be read, and will return
+// a nil value if neither environment variables are set.
+func getGoogleCredentials() (option.ClientOption, error) {
+	if envCredentials, isSet := os.LookupEnv(SopsGoogleCredentialsEnv); isSet {
+		credentials := []byte(envCredentials)
+		if _, err := os.Stat(envCredentials); err == nil {
+			if credentials, err = os.ReadFile(envCredentials); err != nil {
+				return nil, err
+			}
+		}
+		return option.WithCredentialsJSON(credentials), nil
+	}
+	if envToken, isSet := os.LookupEnv(SopsGoogleCredentialsOAuthEnv); isSet {
+		token := []byte(envToken)
+		if _, err := os.Stat(envToken); err == nil {
+			if token, err = os.ReadFile(envToken); err != nil {
+				return nil, err
+			}
+		}
+		tokenSource := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: string(token)},
+		)
+		return option.WithTokenSource(tokenSource), nil
+	}
```

Comment:
I think we should not check for a file here, but rather assume what's documented in https://cloud.google.com/sdk/docs/authorizing. Which only mentions:

> Set the `CLOUDSDK_AUTH_ACCESS_TOKEN` environment variable to the **access token value**.